fdbserver/storageserver.actor.cpp

```diff
@@ -5875,6 +5877,23 @@ ACTOR Future<std::unordered_map<Key, Version>> dispatchChangeFeeds(StorageServer
 	}
 }
 
+namespace {
+ACTOR Future<Version> getRawReadVersion(Transaction* pTransaction) {
+	loop {
+		try {
+			Version readVersion = wait(pTransaction->getRawReadVersion());
+			return readVersion;
+		} catch (Error& err) {
+			if (err.code() != error_code_proxy_memory_limit_exceeded &&
+			    err.code() != error_code_batch_transaction_throttled) {
+				throw err;
+			}
+			wait(delay(SERVER_KNOBS->PROXY_READ_VERSION_BACKOFF));
```

Comment:
Can we use `wait(pTransaction->onError(err));`? Not sure if these two errors are handled correctly, if not, we can use an error that has proper backoff.

Reply:
Also, can we do exponential backoff?

Reply:
I think from the code they are handled properly with a backoff. I will use that in this case.

Reply:
Here I mean `onError`. The current two errors can be handled by this method. If new types of exceptions are thrown from GRV, there could be issues.

Reply:
`onError` does exponential backoff. For now, there are only two types of errors, so it's fine for 7.1, which won't introduce new errors on GRV. In the future, any new errors returned by GRV should be correctly handled here.

Reply:
I would guess it might be better to let `onError` handle the exceptions, since there are more exception-related operations. I do not have a very strong opinion on what errors I can leak here.